icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java

```diff
@@ -786,11 +836,37 @@ public int compareTo(ULocale other) {
 
     /**
      * {@icunote} Unlike the Locale API, this returns an array of <code>ULocale</code>,
-     * not <code>Locale</code>.  Returns a list of all installed locales.
+     * not <code>Locale</code>.
+     *
+     * <p>Returns a list of all installed locales. This is equivalent to calling
+     * {@link #getAvailableLocalesByType} with AvialableType.DEFAULT.
+     *
      * @stable ICU 3.0
      */
     public static ULocale[] getAvailableLocales() {
-        return ICUResourceBundle.getAvailableULocales();
+        return ICUResourceBundle.getAvailableULocales().clone();
+    }
+
+    /**
+     * Returns a list of all installed locales according to the specified type.
+     *
+     * @draft ICU 65
+     * @provisional This API might change or be removed in a future release.
+     */
+    public static Collection<ULocale> getAvailableLocalesByType(AvailableType type) {
+        if (type == null) {
+            throw new IllegalArgumentException();
+        }
+        if (type == ULocale.AvailableType.WITH_LEGACY_ALIASES) {
```

Comment:
Optional: Before this "if", declare a List<ULocale> list. Then in each branch set the list, and after the else body, have a single shared `return Collections.unmodifiableList(list);`.

Reply:
Done.